lib/cfndsl/specification.rb

```diff
@@ -69,7 +69,15 @@ def self.extract_types(spec)
           elsif nested_prop_info['PrimitiveItemType']
             nested_prop_type = Array(nested_prop_info['PrimitiveItemType'])
           elsif nested_prop_info['ItemType']
-            nested_prop_type = Array(root_resource_name + nested_prop_info['ItemType'])
+            # Tag is a reused type, but not quite primitive
```

Comment:
This is my fix for #370

Reply:
I was thinking the other day there must be a smarter way to solve this. This should mean we can get rid of https://github.com/cfndsl/cfndsl/blob/master/lib/cfndsl/patches.rb#L71-L94